### PR TITLE
WIP: IMCA Shipments

### PR DESCRIPTION
**Pull request by stufisher**
Adds:

* Generic profile based CSV importer see imca.js
* Mobile container assigner based on barcode scanning
* Ability to show diffraction plan columns when registering a container (show DP columns)

And the two db changes:
https://github.com/DiamondLightSource/ispyb-database/pull/46

https://github.com/DiamondLightSource/ispyb-database/pull/43
https://github.com/DiamondLightSource/ispyb-database/pull/47

### CODE REVIEW

**Comment by jlmuir** [comment]
If I want to export a shipment to CSV from the Shipment view, I have to click the "Import from CSV" button which takes me to the "Import from CSV" view where there is an "Export to CSV" button that I can press.  This is not intuitive (i.e., clicking the "import" button when I want to "export").  It would be better if there were another button on the Shipment view called "Export to CSV" right next to the "Import from CSV" button.  Or maybe easier would be to change the label of the "Import from CSV" button to "CSV Import/Export"?  Or maybe some other approach that I haven't thought of.

shipment-import-from-csv-button

**Comment by stufisher** [comment]
> If I want to export a shipment to CSV from the Shipment view, I have to click the "Import from CSV" button which takes me to the "Import from CSV" view where there is an "Export to CSV" button that I can press. This is not intuitive (i.e., clicking the "import" button when I want to "export"). It would be better if there were another button on the Shipment view called "Export to CSV" right next to the "Import from CSV" button. Or maybe easier would be to change the label of the "Import from CSV" button to "Import/Export CSV" or "CSV Import/Export"? Or maybe some other approach that I haven't thought of.
> 
> shipment-import-from-csv-button

Resolved in https://github.com/DiamondLightSource/SynchWeb/pull/136/commits/3feb552f7c97a61f3b82632efc3e890d97f523ca

File: client/src/js/modules/shipment/views/shipment.js [hunk 1]
```diff
@@ -43,7 +43,9 @@ define(['marionette',
                 APIURL: app.apiurl,
                 PROP: app.prop,
                 DHL_ENABLE: app.options.get('dhl_enable'),
-                IS_STAFF: app.staff
+                IS_STAFF: app.staff,
+                QUEUE_SHIPMENT: app.options.get('queue_shipment'),
```
Comment (jlmuir) [inline client/src/js/modules/shipment/views/shipment.js:47]: This should be

```
QUEUE_SHIPMENT: app.config.queue_shipment,
```

Comment (stufisher) [reply client/src/js/modules/shipment/views/shipment.js]: Done

File: client/src/js/csv/imca.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        // Export transforms
+        export: {
+            CELL: function(m) {
+                return `${m.CELL_A}, ${m.CELL_B}, ${m.CELL_C}, ${m.CELL_ALPHA}, ${m.CELL_BETA}, ${m.CELL_GAMMA}`.trim()
+            },
+
+            STATUS: function(m) {
+                var status = 'skipped'
+                if (m.QUEUEDTIMESTAMP) status = 'queued';
+                if (m.R > 0) status = 'recieved'
```
Comment (jlmuir) [inline client/src/js/csv/imca.js:61]: This is a typo: `recieved` should be `received`.

**Comment by ndg63276** [comment]
@stufisher - can this PR be closed? It looks like https://github.com/DiamondLightSource/SynchWeb/pull/239 supersedes it?

**Pull request closed without merging**